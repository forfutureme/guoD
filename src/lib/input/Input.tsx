/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-13 08:40:49
 * @Desc: Input 组件
 */
import React from 'react'
import './input.scss'
import {prefixClass, joinClassName} from '@utils/classNameHandle'
const fixClass = prefixClass('guo-input')
interface props extends React.InputHTMLAttributes<HTMLInputElement> {
  count?: boolean
  type?: string
}
const Input: React.FC<props> = ({
  className,
  type = 'text',
  count = false,
  value,
  maxLength,
  ...rest
}) => {
  return (
    <div className={fixClass('box')}>
      <input
        className={joinClassName(fixClass(), className)}
        type={type}
        value={value}
        maxLength={maxLength}
        {...rest}
      />
      {count && maxLength && (
        <span className={fixClass('num')}>
          {value.toString().length}/{maxLength}
        </span>
      )}
    </div>
  )
}
export default Input
