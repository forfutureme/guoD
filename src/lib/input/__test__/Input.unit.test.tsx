/**
 * @format
 * @Author: huweijian
 * @Date: 2020-02-14 08:12:23
 * @Desc: Input组件单元测试
 */
import renderer from 'react-test-renderer'
import React from 'react'

import Input from '../Input'

describe('Input测试', () => {
  it('测试基本的input渲染', () => {
    const json = renderer.create(<Input />)
    expect(json).toMatchSnapshot()
  })
  it('测试限制输入长度的input', () => {
    const json = renderer.create(<Input maxLength={10} />)
    expect(json).toMatchSnapshot()
  })
  it('测试限制输入长度同时显示已输入字数的input', () => {
    const json = renderer.create(
      <Input maxLength={10} count={true} value="1" />
    )
    expect(json).toMatchSnapshot()
  })
})
