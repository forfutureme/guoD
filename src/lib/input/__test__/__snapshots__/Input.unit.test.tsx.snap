// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input测试 测试基本的input渲染 1`] = `
<div
  className="guo-input-box"
>
  <input
    className="guo-input"
    type="text"
  />
</div>
`;

exports[`Input测试 测试限制输入长度同时显示已输入字数的input 1`] = `
<div
  className="guo-input-box"
>
  <input
    className="guo-input"
    maxLength={10}
    type="text"
    value="1"
  />
  <span
    className="guo-input-num"
  >
    1
    /
    10
  </span>
</div>
`;

exports[`Input测试 测试限制输入长度的input 1`] = `
<div
  className="guo-input-box"
>
  <input
    className="guo-input"
    maxLength={10}
    type="text"
  />
</div>
`;
